x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -308,7 +303,33 @@ public void deleteEngine(String engineName, ActionListener<DeleteResponse> liste
         } catch (Exception e) {
             listener.onFailure(e);
         }
+    }
+
+    GetAliasesResponse getAlias(String engineAliasName) {
+        return clientWithOrigin.admin().indices().getAliases(new GetAliasesRequest(engineAliasName)).actionGet();
+    }
+
+    private void removeAlias(String engineAliasName, ActionListener<AcknowledgedResponse> listener) {
+        IndicesAliasesRequest aliasesRequest = new IndicesAliasesRequest().addAliasAction(
+            IndicesAliasesRequest.AliasActions.remove().aliases(engineAliasName).indices("*")
+        );
+        clientWithOrigin.admin()
+            .indices()
+            .aliases(aliasesRequest, listener.delegateFailure((l, r) -> l.onResponse(AcknowledgedResponse.TRUE)));
+    }
+
+    public void deleteEngineAndAlias(String engineName, ActionListener<DeleteResponse> listener) {
+        deleteEngine(engineName, new ActionListener<>() {
```

Comment:
@carlosdelest I didn't see a great way to do that, could you please offer an example?

Reply:
Sure! It would look like:

```java
    public void deleteEngineAndAlias(String engineName, ActionListener<DeleteResponse> listener) {
        removeAlias(Engine.getEngineAliasName(engineName), listener.delegateFailure((l, r) -> deleteEngine(engineName, l)));
    }
```

Reply:
Thanks, changed!